src/Microsoft.OData.Core/ODataNotificationWriter.cs

```diff
@@ -15,19 +15,25 @@ namespace Microsoft.OData
     /// <summary>
     /// Wrapper for TextWriter to listen for dispose.
     /// </summary>
+#if NETSTANDARD2_0
+    internal sealed class ODataNotificationWriter : TextWriter, IAsyncDisposable
+#else
     internal sealed class ODataNotificationWriter : TextWriter
+#endif
     {
         private readonly TextWriter textWriter;
         private IODataStreamListener listener;
+        private bool synchronous;
 
-        internal ODataNotificationWriter(TextWriter textWriter, IODataStreamListener listener)
+        internal ODataNotificationWriter(TextWriter textWriter, IODataStreamListener listener, bool synchronous = true)

```

Comment:
Was there another way to implement this without adding the `bool synchronous` parameter?

Reply:
I couldn't figure any other way. If we were only supporting .NET Core 3.1 or higher it'd be easy because in asynchronous situations we'd always be guaranteed of `DisposeAsync`. But for say .NET FX and .NET Core 1.x or 2.x, we honestly have no way of know from the `Dispose` method whether we're doing things synchronously or asynchronously. That's where the `synchronous` parameter comes in.